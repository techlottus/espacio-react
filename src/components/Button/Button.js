

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


