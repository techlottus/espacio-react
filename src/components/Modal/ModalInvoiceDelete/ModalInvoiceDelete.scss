@import '../../../App.scss';
@import '../Modal.scss';

.modal-invoice-delete {
  @extend .modal-container;

}