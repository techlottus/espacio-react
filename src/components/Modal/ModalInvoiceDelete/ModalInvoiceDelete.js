import React, { useEffect, useRef, useState } from "react";
import { Button } from "../../Button/Button";
import {
  btnModalBackInvoice,
  btnModalNextInvoice,
} from "../../../constants/Payment.constant";
import "./ModalInvoiceDelete.scss";
import { useSelector } from "react-redux";

export const ModalInvoiceDelete = ({ isShow, onClose, saveModal, outside }) => {
  const { texts } = useSelector(state => state.texts);
  const [isModal, setIsModal] = useState(false);

  const modalRef = useRef();

  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]);

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-invoice-delete">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
          <span className="material-icons icon iconright">{texts?.payment.modals.modalIcon}</span>
            <div className="title">{texts?.payment.modals.modalInvoiceDelete.title}</div>
          </div>
          <div className="modalbody">
            <div className="modalbtns">
              <div className="btnsecundary">
                <Button
                  data={{
                    ...btnModalBackInvoice,
                    title: texts?.payment.btnModal.back,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    onClose();
                  }}
                />
              </div>
              <div className="btnprimary">
                <Button
                  data={{
                    ...btnModalNextInvoice,
                    title: texts?.payment.btnModal.next,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    saveModal()
                    onClose();
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
