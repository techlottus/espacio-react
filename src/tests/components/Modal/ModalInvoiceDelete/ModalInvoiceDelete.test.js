import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ModalInvoiceDelete } from "../../../../components/Modal/ModalInvoiceDelete/ModalInvoiceDelete";
import { act } from "react-dom/test-utils";
import { Button } from "../../../../components/Button/Button";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
    texts: {},
}

const isShow = true;
const onClose = jest.fn();
const saveModal = true;
const outside = jest.fn();


let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <ModalInvoiceDelete isShow={isShow} onClose={onClose} saveModal={saveModal} outside={outside}/>
    </Provider>
  );
};

describe("Test <ModalInvoiceDelete/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test("should close modal", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Button).first().props().onClick();
    });
    wrapper.update();
  });


});
