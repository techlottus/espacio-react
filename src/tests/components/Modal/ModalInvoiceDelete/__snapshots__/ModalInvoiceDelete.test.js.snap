// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ModalInvoiceDelete/> should init component 1`] = `
<div
  className="modal-invoice-delete"
>
  <div
    className="modalcontent"
  >
    <div
      className="modalheader"
    >
      <div
        className="title"
      />
    </div>
    <div
      className="modalbody"
    >
      <div
        className="modalbtns"
      >
        <div
          className="btnsecundary"
        >
          <lottus-button />
        </div>
        <div
          className="btnprimary"
        >
          <lottus-button />
        </div>
      </div>
    </div>
  </div>
</div>
`;
